miden-lib/asm/miden/kernels/tx/memory.masm

```diff
@@ -1069,7 +1073,13 @@ end
 #! - note_ptr is the memory address at which the created note data begins.
 #! - num_assets is the number of assets in the created note.
 export.set_created_note_num_assets
-    push.CREATED_NOTE_NUM_ASSETS_OFFSET add mem_store
+    push.CREATED_NOTE_NUM_ASSETS_OFFSET add
+    # => [note_ptr + offset, num_assets]
+
+    # check note number of assets limit
+    dup.1 exec.constants::get_max_assets_per_note gt assert.err=ERR_NOTE_TOO_MANY_ASSETS
```

Comment:
Let's also update the comments for the function to indicate when the function panics.